### ISSUE TITLE
test_anon_download (Tribler.Test.test_tunnel_community.TestTunnelCommunity) blocks the wx thread shortly after starting

### ISSUE THREAD

**Issue by whirm** [Tribler/tribler#1418]
And just a bit later the twisted reactor blocks too.

It doesn't always happen on the PR unit test run, but I can reproduce it with 100% success locally.

I'm working on instrumenting the base test classes to check for this and fail early/report with debugging info when it happens.

I'm not sure if it's due to the recent changes made by @NielsZeilemaker and @LipuFei  or not, as I tried running the remote search test set and a few others locally a couple times and they seem to work just fine.

**Comment by NielsZeilemaker** [comment]
Is the NoDispersyRLock still there? If not, revert that change and see if
you can find where it's deadlocking.
Its probably the wxthread acquiring the lock, and then the twisted thread
attempting to acquire it.


2015-05-13 18:44 GMT+02:00 Elric Milon <notifications@github.com>:

**Comment by whirm** [comment]
Quick note: I got this one stuck too in an isolated run:
```
test_anon_download_without_exitnodes (Tribler.Test.test_tunnel_community.TestTunnelCommunity) ```

**Comment by whirm** [comment]
@rjruigrok so far it seems to only happen in the tunneled tests?

**Comment by whirm** [comment]
@rjruigrok has this happened to you lately on the current tunnel PR?

**Comment by rjruigrok** [comment]
Not sure, most of the tests in my PR are totally broken. But the latest ones all should have passed. 
Is http://jenkins.tribler.org//job/GH_Tribler_pull-request-tester-turbo-isolated_devel/451/ stuck?

**Comment by whirm** [comment]
How to tell if it timed out: 
* If there's no test results, it probably timed out.
So we look at the job output: http://jenkins.tribler.org/job/GH_Tribler_pull-request-tester-turbo-isolated_devel/451/artifact/output/
```
	1_nosetests.xml.part	1.26 KB	 view
	2_nosetests.xml.part	1.75 KB	 view
	4_nosetests.xml.part	2.17 KB	 view
	5_nosetests.xml.part	1.01 KB	 view
	6_nosetests.xml.part	5.22 KB	 view
	7_nosetests.xml.part	1.59 KB	 view
	8_nosetests.xml.part	984 B	 view
	9_nosetests.xml.part	757 B	 view
```
The test report for group 3 is missing, so that means that it either crashed badly or that it got killed due to a timeout.

Let's go to it's stderr file: http://jenkins.tribler.org/job/GH_Tribler_pull-request-tester-turbo-isolated_devel/451/artifact/output/test_runners_output/00003.err/*view*/

And we can see it got stuck at: 
```
test_anon_download (Tribler.Test.test_tunnel_community_positive.TestTunnelCommunityPositive) ...
```

**Comment by rjruigrok** [comment]
I understand. But is this stuck test related to the blocking wx thread? 
These stuck tests occasionly occur, but the timeouts are narrowed in the previous PR, I don't really know what to do about it, or what to fix, as there is no log or traceback.

**Comment by whirm** [comment]
I don't know if this one in particular is. I have a branch which adds a watchdog on both WX and twisted threads. I'll see if I can integrate it so when one of the watchdog trips, the test fails too.

**Comment by whirm** [comment]
The watchdog branch has been merged.

**Comment by whirm** [comment]
BTW: the watchog won't make the tests fail (it wouldn't work most of the time as if the twisted thread is locked, the test-ending checks will not be made anymore) We could play with injecting exceptions on the stuck thread, etc. but I don't think it's worth the trouble as it wouldn't work in all situations anyway.

**Comment by whirm** [comment]
@LipuFei  @NielsZeilemaker @synctext ^ have a look at my last 4 comments about the watchdog thing.

**Comment by whirm** [comment]
Closed by #1510
